metadata.go

```diff
@@ -19,16 +19,30 @@ package ttrpc
 import (
 	"context"
 	"strings"
+	"sync"
 )
 
 // MD is the user type for ttrpc metadata
-type MD map[string][]string
+type MD struct {
+	data map[string][]string
+	mu   sync.RWMutex
+}
```

Comment:
Is this suppose to be used like a `http.Header` where you need to `Clone` when sharing with other "requests", or suppose to be shared and need a mutex?

Reply:
I think both ways are fine and we just need to stick with one, I just think the change will be easier to operate with the mutex solution given that there are MD objects that are mutated with the `Set` and `Append` (and `setRequest`) functions (like [here](https://github.com/containerd/ttrpc/blob/b71d9dee113bc6d5d5479aab0f58fb459ae24f73/client.go#L163C12-L163C22))